### PR TITLE
F-71A fixes for uid mismatch and capacity function

### PR DESCRIPTION
**Pull request by aditipande96**
Signed-off-by: Aditi Pande <aditi.pande@seagate.com>

# Problem Statement
- Problem statement

# Design
-  For Bug, Describe the fix here.
-  For Feature, Post the link for design

# Coding
   Checklist for Author
-  [ ] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
-  [ ] New/Affected tests are executed on Latest Build
-  [ ] Attach test execution logs
-  [ ] Collection tested and no collection error introduced (`pytest --local True --collect-only`)

# Review Checklist 
  Checklist for Author
- [ ] JIRA number/GitHub Issue added to PR
- [ ] PR is self reviewed
- [ ] Jira and state/status is updated and JIRA is updated with PR link
- [ ] Check if the description is clear and explained

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
-  [ ] If change in any common function, make sure to update all calls and execute all affected tests.

# Documentation
  Checklist for Author
-  [ ] Changes done to ReadMe / WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

**Comment by aditipande96** [comment]
Collections are green:
====== 2336 tests collected in 14.38s ========

f-71a-status

File: tests/csm/rest/test_capacity_quota.py [hunk 1]
```diff
@@ -394,16 +394,19 @@ def test_40636(self):
         test_case_name = cortxlogging.get_frame()
         self.log.info("##### Test started -  %s #####", test_case_name)
         self.log.info("Step 1: Creating 2 IAM user with same uid but different tenant")
+        uid = "iam_user_1_" + str(int(time.time()))
```
Comment (kachhwahadivya) [inline tests/csm/rest/test_capacity_quota.py:397]: This should be inside for loop.

File: libs/csm/rest/csm_rest_quota.py [hunk 2]
```diff
@@ -252,8 +252,8 @@ def verify_user_capacity(self, user_id, used:int, used_rounded:int=None, obj_cnt
             self.log.info("Expected: %s", user_id)
 
             if user_id != auser_id:
-                self.log.info("User ID mismatch")
-                result = False
+                self.log.error("User ID mismatch")
+                #result = False
```
Comment (kachhwahadivya) [inline libs/csm/rest/csm_rest_quota.py:256]: Can u uncomment this?
We dont want false positive on main.

File: tests/csm/rest/test_capacity_quota.py [hunk 3]
```diff
@@ -786,20 +790,9 @@ def test_41127(self):
         self.log.info("total objects and size %s and %s ", total_objects, total_size)
         self.log.info("Data size is %s ", data_size)
         self.log.info("Step 4: Perform & Verify GET API to get capacity usage stats")
```
Comment (kachhwahadivya) [inline tests/csm/rest/test_capacity_quota.py:792]: Can u correct skip marker for [TEST-40640]

File: libs/csm/rest/csm_rest_quota.py (hunk 2 above)
Comment (aditipande96) [reply libs/csm/rest/csm_rest_quota.py]: done

File: tests/csm/rest/test_capacity_quota.py (hunk 3 above)
Comment (aditipande96) [reply tests/csm/rest/test_capacity_quota.py:792]: done

File: tests/csm/rest/test_capacity_quota.py [hunk 4]
```diff
@@ -1846,19 +1735,9 @@ def test_43834(self):
...
-        uid = resp.json()["capacity"]["s3"]["users"][0]["id"]
-        t_obj = resp.json()["capacity"]["s3"]["users"][0]["objects"]
-        t_size = resp.json()["capacity"]["s3"]["users"][0]["used"]
-        m_size = resp.json()["capacity"]["s3"]["users"][0]["used_rounded"]
-
-        assert_utils.assert_equals(self.uid, uid, "id is not equal")
-        assert_utils.assert_equals(total_objects, t_obj, "Number of objects not equal")
-        assert_utils.assert_equal(total_size, t_size, "Total Size mismatch found")
-        assert_utils.assert_greater_equal(m_size, total_size, "Total Used Size mismatch found ")
-        assert_utils.assert_greater(m_size, t_size, "used_rounded and used mismatch found ")
```
Comment (RajeshPratapSInghChouhan) [inline tests/csm/rest/test_capacity_quota.py:1861]: same as above, we need to have the last asset, that is not verified in verify_user_capacity

File: tests/csm/rest/test_capacity_quota.py [hunk 5]
```diff
@@ -1900,20 +1779,10 @@ def test_43835(self):
...
-        uid = resp.json()["capacity"]["s3"]["users"][0]["id"]
-        t_obj = resp.json()["capacity"]["s3"]["users"][0]["objects"]
-        t_size = resp.json()["capacity"]["s3"]["users"][0]["used"]
-        m_size = resp.json()["capacity"]["s3"]["users"][0]["used_rounded"]
-
-        assert_utils.assert_equals(self.uid, uid, "id is not equal")
-        assert_utils.assert_equals(total_objects, t_obj, "Number of objects not equal")
-        assert_utils.assert_equal(total_size, t_size, "Total Size mismatch found")
-        assert_utils.assert_greater_equal(m_size, total_size, "Total Used Size mismatch found ")
-        assert_utils.assert_equal(m_size, t_size, "used_rounded and used mismatch found ")
```
Comment (RajeshPratapSInghChouhan) [inline tests/csm/rest/test_capacity_quota.py:1916]: same as above, we need to have the last asset, that is not verified in verify_user_capacity

File: tests/csm/rest/test_capacity_quota.py [hunk 6]
```diff
@@ -1958,19 +1827,9 @@ def test_43836(self):
...
```
Comment (RajeshPratapSInghChouhan) [inline tests/csm/rest/test_capacity_quota.py:1973]: same as above, we need to have the last asset, that is not verified in verify_user_capacity

File: tests/csm/rest/test_capacity_quota.py [hunk 7]
```diff
@@ -2157,21 +1995,9 @@ def test_43839(self):
...
-        uid = resp.json()["capacity"]["s3"]["users"][0]["id"]
-        t_obj = resp.json()["capacity"]["s3"]["users"][0]["objects"]
-        t_size = resp.json()["capacity"]["s3"]["users"][0]["used"]
-        m_size = resp.json()["capacity"]["s3"]["users"][0]["used_rounded"]
-
-        assert_utils.assert_equals(self.uid, uid, "id is not equal")
-        assert_utils.assert_equals(total_objects, t_obj, "Number of objects not equal")
-        assert_utils.assert_equal(total_size, t_size, "Total Size mismatch found")
-        assert_utils.assert_greater_equal(m_size, total_size, "Total Used Size mismatch found ")
-        assert_utils.assert_greater_equal(m_size, t_size, "used_rounded and used mismatch found ")
```
Comment (RajeshPratapSInghChouhan) [inline tests/csm/rest/test_capacity_quota.py:2172]: same as above, we need to have the last asset, that is not verified in verify_user_capacity

File: tests/csm/rest/test_capacity_quota.py [hunk 8]
```diff
@@ -2264,20 +2090,9 @@ def test_43840(self):
...
```
Comment (RajeshPratapSInghChouhan) [inline tests/csm/rest/test_capacity_quota.py:2279]: same as above, we need to have this asset, that is not verified in verify_user_capacity

File: tests/csm/rest/test_capacity_quota.py (hunk 4 above)
Comment (aditipande96) [reply tests/csm/rest/test_capacity_quota.py:1861]: done

File: tests/csm/rest/test_capacity_quota.py (hunk 5 above)
Comment (aditipande96) [reply tests/csm/rest/test_capacity_quota.py:1916]: done

File: tests/csm/rest/test_capacity_quota.py [hunk 9]
```diff
@@ -2069,19 +1918,9 @@ def test_43838(self):
...
-        uid = resp.json()["capacity"]["s3"]["users"][0]["id"]
-        t_obj = resp.json()["capacity"]["s3"]["users"][0]["objects"]
-        t_size = resp.json()["capacity"]["s3"]["users"][0]["used"]
-        m_size = resp.json()["capacity"]["s3"]["users"][0]["used_rounded"]
-
-        assert_utils.assert_equals(self.uid, uid, "id is not equal")
-        assert_utils.assert_equals(total_objects, t_obj, "Number of objects not equal")
-        assert_utils.assert_equal(total_size, t_size, "Total Size mismatch found")
-        assert_utils.assert_greater_equal(m_size, total_size, "Total Used Size mismatch found ")
-        assert_utils.assert_equal(m_size, t_size, "used_rounded and used mismatch found ")
```
[the remark this answers is not in the document]
Comment (aditipande96) [reply tests/csm/rest/test_capacity_quota.py:2084]: done

File: tests/csm/rest/test_capacity_quota.py (hunk 7 above)
Comment (aditipande96) [reply tests/csm/rest/test_capacity_quota.py:2172]: done

File: tests/csm/rest/test_capacity_quota.py [hunk 10]
```diff
@@ -2157,21 +1995,9 @@ def test_43839(self):
...
-        t_obj = resp.json()["capacity"]["s3"]["users"][0]["objects"]
-        t_size = resp.json()["capacity"]["s3"]["users"][0]["used"]
-        m_size = resp.json()["capacity"]["s3"]["users"][0]["used_rounded"]
-
-        assert_utils.assert_equals(self.uid, uid, "id is not equal")
-        assert_utils.assert_equals(total_objects, t_obj, "Number of objects not equal")
-        assert_utils.assert_equal(total_size, t_size, "Total Size mismatch found")
-        assert_utils.assert_greater_equal(m_size, total_size, "Total Used Size mismatch found ")
-        assert_utils.assert_greater_equal(m_size, t_size, "used_rounded and used mismatch found ")
-        #assert_utils.assert_greater_equal(used_data_size, m_size) Revisit after CORTX-32486
```
[the remark this answers is not in the document]
Comment (aditipande96) [reply tests/csm/rest/test_capacity_quota.py:2173]: done

File: tests/csm/rest/test_capacity_quota.py [hunk 11]
```diff
@@ -2264,20 +2090,9 @@ def test_43840(self):
...
```
[the remark this answers is not in the document]
Comment (aditipande96) [reply tests/csm/rest/test_capacity_quota.py:2280]: done

**Review by RajeshPratapSInghChouhan — APPROVED** [review]

**Review by kachhwahadivya — APPROVED** [review]

**Review by ravindrachoudhari — APPROVED** [review]

**Comment by aditipande96** [comment]
Collection Logs:
========= 2336 tests collected in 15.76s ========

**Pull request merged**